buildtools/webpack.config.js

```diff
@@ -71,6 +71,12 @@ module.exports = {
       { test: /\.js$/,
         use: ["source-map-loader"],
         enforce: "pre"
+      },
+      {
+        test: /\.m?js/,
+        resolve: {
+          fullySpecified: false
+        }
```

Comment:
I admit this is a piece of code I found on the internet to fix issues while building with Sinon upgrade, and that I don't understand fully neither their consequences.

I can try to know more about that next week.

Reply:
Do you happen to have a log of what those issues were?

Reply:
Yes, here are the logs when I run `yarn build:prod` and without the above patch. Removing the `node_modules/` folder doesn't solve anything.

```
yarn run v1.22.21
$ buildtools/build.sh
+ tsc --build
+ buildtools/update_type_info.sh app
+ webpack --config buildtools/webpack.config.js --mode production
assets by status 8.11 MiB [cached] 21 assets
orphan modules 1.13 MiB [orphan] 236 modules
runtime modules 20.6 KiB 46 modules
modules by path ./node_modules/ 8.18 MiB
  javascript modules 7.86 MiB 652 modules
  json modules 327 KiB 3 modules
modules by path ./app/ 3.87 MiB 279 modules
modules by path ./test/ 219 KiB 28 modules
modules by path ./stubs/app/ 671 bytes
  modules by path ./stubs/app/client/ui/*.ts 386 bytes 3 modules
  ./stubs/app/common/version.ts 156 bytes [built] [code generated]
  ./stubs/app/client/components/Banners.ts 129 bytes [built] [code generated]
external "jQuery" 42 bytes [built] [code generated]
external "alert" 42 bytes [built] [code generated]
./package.json 6.45 KiB [built] [code generated]

WARNING in ./stubs/app/common/version.ts 2:23-42
Should not import the named export 'version' (imported as 'packageJson') from default-exporting module (only default export is available soon)
 @ ./app/client/ui/AppHeader.ts 5:0-46 40:170-185 40:191-208 40:230-247
 @ ./app/client/ui/errorPages.ts 3:0-52 90:25-34
 @ ./app/client/errorMain.ts 1:0-56 2:0-12

ERROR in ./node_modules/sinon/pkg/sinon-esm.js 5166:15-22
Module not found: Error: Can't resolve 'process/browser' in '/home/florentpro/projects/grist-core/node_modules/sinon/pkg'
Did you mean 'browser.js'?
BREAKING CHANGE: The request 'process/browser' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
resolve 'process/browser' in '/home/florentpro/projects/grist-core/node_modules/sinon/pkg'
  Parsed request is a module
  using description file: /home/florentpro/projects/grist-core/node_modules/sinon/package.json (relative path: ./pkg)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in /home/florentpro/projects/grist-core
        /home/florentpro/projects/grist-core/process doesn't exist
      looking for modules in /home/florentpro/projects/grist-core/ext
        /home/florentpro/projects/grist-core/ext/process doesn't exist
      looking for modules in /home/florentpro/projects/grist-core/stubs
        /home/florentpro/projects/grist-core/stubs/process doesn't exist
      looking for modules in /home/florentpro/projects/grist-core/node_modules
        existing directory /home/florentpro/projects/grist-core/node_modules/process
          using description file: /home/florentpro/projects/grist-core/node_modules/process/package.json (relative path: .)
            using description file: /home/florentpro/projects/grist-core/node_modules/process/package.json (relative path: ./browser)
              Field 'browser' doesn't contain a valid alias configuration
              /home/florentpro/projects/grist-core/node_modules/process/browser doesn't exist
 @ ./test/common/InactivityTimer.ts 4:0-31 8:10-19
 @ ./test/client-harness/client.js

webpack 5.73.0 compiled with 1 error and 1 warning in 37312 ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
I struggled a bit again, but here is what I hope would work: https://github.com/gristlabs/grist-core/pull/856/files#diff-249979d31b349e184d0568ed390ed1ad3fd8ca63e59b9f0c38fcde4d9ec445c5